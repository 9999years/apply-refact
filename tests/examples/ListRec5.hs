foos [] x = x; foos (y:ys) x = foo y $ foos ys x