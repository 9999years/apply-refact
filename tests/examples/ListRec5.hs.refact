[("tests/examples/ListRec5.hs:1:1-48: Suggestion: Use foldr\nFound:\n  foos [] x = x\n  foos (y : ys) x = foo y $ foos ys x\nPerhaps:\n  foos ys x = foldr foo x ys\n",[Replace {rtype = Decl, pos = SrcSpan {startLine = 1, startCol = 1, endLine = 1, endCol = 49}, subts = [], orig = "foos ys x = foldr foo x ys"}])]
