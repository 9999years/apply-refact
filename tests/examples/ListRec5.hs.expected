foos ys x = foldr foo x ys
